src/main/kotlin/com/terraformation/backend/seedbank/api/AccessionsV2Controller.kt

```diff
@@ -0,0 +1,368 @@
+import com.terraformation.backend.db.DataSource
+import com.terraformation.backend.db.FacilityId
+import com.terraformation.backend.db.ProcessingMethod
+import com.terraformation.backend.db.SpeciesId
+import com.terraformation.backend.db.StorageCondition
+import com.terraformation.backend.seedbank.db.AccessionStore
+import com.terraformation.backend.seedbank.model.AccessionActive
+import com.terraformation.backend.seedbank.model.AccessionModel
+import com.terraformation.backend.seedbank.model.Geolocation
+import com.terraformation.backend.seedbank.model.isV2Compatible
+import com.terraformation.backend.seedbank.model.toV2Compatible
+import com.terraformation.backend.util.orNull
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import java.time.Clock
+import java.time.LocalDate
+import javax.validation.Valid
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RequestMapping("/api/v2/seedbank/accessions")
+@RestController
+@SeedBankAppEndpoint
+class AccessionsV2Controller(
+    private val accessionStore: AccessionStore,
+    private val clock: Clock,
+) {
+  @ApiResponse(
+      responseCode = "200",
+      description =
+          "The accession was created successfully. Response includes fields populated by the " +
+              "server, including the accession number and ID.")
+  @Operation(summary = "Creates a new accession.")
+  @PostMapping
+  fun createAccession(
+      @RequestBody payload: CreateAccessionRequestPayloadV2
+  ): CreateAccessionResponsePayloadV2 {
+    val updatedPayload = accessionStore.create(payload.toModel())
+    return CreateAccessionResponsePayloadV2(AccessionPayloadV2(updatedPayload))
+  }
+
+  @ApiResponse(
+      responseCode = "200",
+      description =
+          "The accession was updated successfully. Response includes fields populated or " +
+              "modified by the server as a result of the update.")
+  @ApiResponse404(description = "The specified accession doesn't exist.")
+  @Operation(summary = "Update an existing accession.")
+  @PutMapping("/{id}")
+  fun updateAccession(
+      @RequestBody payload: UpdateAccessionRequestPayloadV2,
+      @PathVariable("id") accessionId: AccessionId,
+      @RequestParam
+      @Schema(
+          description =
+              "If true, do not actually save the accession; just return the result that would " +
+                  "have been returned if it had been saved.")
+      simulate: Boolean?
```

Comment:
not sure why my commend didn't come through... would the UI client need to use this for the near term until feature is fully implemented?

Reply:
The API should be safe to call for real immediately. I can remove the `simulate` parameter if the client is never actually using it. At one point we needed it in order to do some client-side validation or to preview some calculated values, but I don't remember the specifics.